src/encord_active/lib/common/active_learning.py

```diff
@@ -0,0 +1,91 @@
+# Temporary place for utility functions used in active learning examples.
+# They should be integrated as features of the corresponding classes.
+import json
+
+import pandas as pd
+
+from encord_active.lib.common.iterator import DatasetIterator
+from encord_active.lib.metrics.acquisition_functions import BaseModelWrapper
+from encord_active.lib.project.project_file_structure import ProjectFileStructure
+
+
+def get_data(
+    project_fs: ProjectFileStructure, model: BaseModelWrapper, data_hashes: list[tuple[str, str]], class_name: str
+):
+    image_paths, y = get_data_from_data_hashes(project_fs, data_hashes, class_name)
+    X = model.prepare_data(image_paths)
+    return X, y
+
+
+def get_data_hashes_from_project(project_fs: ProjectFileStructure, subset_size: int = None) -> list[tuple[str, str]]:
+    iterator = DatasetIterator(project_fs.project_dir, subset_size)
+    data_hashes = [(iterator.label_hash, iterator.du_hash) for _ in iterator.iterate()]
+    return data_hashes
+
+
+def get_data_from_data_hashes(project_fs: ProjectFileStructure, data_hashes: list[tuple[str, str]], class_name: str):
+    image_paths, class_labels = zip(*(get_data_sample(project_fs, data_hash, class_name) for data_hash in data_hashes))
+    return list(image_paths), list(class_labels)
+
+
+def get_data_sample(project_fs: ProjectFileStructure, data_hash: tuple[str, str], class_name: str):
+    label_hash, du_hash = data_hash
+    lr_struct = project_fs.label_row_structure(label_hash)
+
+    # get classification label
+    label_row = json.loads(lr_struct.label_row_file.read_text())
+    class_label = get_classification_label(label_row, du_hash, class_name=class_name)
+
+    # get image path
+    image_path = lr_struct.images_dir / f"{du_hash}.{label_row['data_units'][du_hash]['data_type'].split('/')[-1]}"
```

Comment:
```suggestion
    image_path = next((dus.path for dus in lr_struct.iter_data_unit()))
```
This avoids data misses - as mentioned, we should always use the structures now. No manual path creation. It will fail.

Reply:
fixed